src/main/java/org/jboss/forge/arquillian/command/cube/CubeSetupCommand.java

```diff
@@ -0,0 +1,336 @@
+            if (source == null) {
+                return null;
+            }
+            if (builder.getUIContext().getProvider().isGUI()) {
+                return source;
+            }
+            return getStringForCLIDisplay(source);
+        });
+
+        filePath.setEnabled(true);
+
+        dockerMachine.setEnabled(this::isDockerOrDockerCompose);
+        dockerMachine.setRequired(this::isDockerOrDockerCompose);
+
+        machineName.setEnabled(this::isDockerMachine);
+        machineName.setRequired(this::isDockerMachine);
+
+        dockerFileName.setEnabled(this::isDocker);
+    }
+
+    @Override
+    public UICommandMetadata getMetadata(UIContext context) {
+        return Metadata.from(super.getMetadata(context), getClass())
+            .category(Categories.create("Arquillian"))
+            .name("Arquillian: Cube Setup")
+            .description("This addon will help you setup Arquillian Cube.");
+    }
+
+    @Override
+    public Result execute(UIExecutionContext context) throws Exception {
+
+        final String msg = checkResourcesExists(context);
+
+        if (type.hasValue()) {
+            setCubeSetupFacet();
+            setCubeConfigurationParamteres(context);
+        }
+
+        if (msg != null) {
+            return Results.fail(msg);
+        }
+
+        final CubeConfiguration cubeConfiguration = cubeSetupFacet.getCubeConfiguration();
+
+        try {
+            facetFactory.install(getSelectedProject(context), cubeSetupFacet);
+
+            return Results.success("Installed Cube " + cubeConfiguration.getType() + " & updated arquillian configuration.");
+        } catch (Exception e) {
+            return Results.fail("Could not install Cube " + cubeConfiguration.getType(), e);
+        }
+    }
+
+    private void setCubeSetupFacet() {
+
+        CubeSetupFacet cubeSetupFacet = new CubeSetupFacet();
+        if (isKubernetes()) {
+            cubeSetupFacet.setCubeConfiguration(CubeConfiguration.KUBERNETES);
+        } else if (isDocker()) {
+            cubeSetupFacet.setCubeConfiguration(CubeConfiguration.DOCKER);
+        } else if (isDockerCompose()) {
+            cubeSetupFacet.setCubeConfiguration(CubeConfiguration.DOCKER_COMPOSE);
+        } else if (isOpenshift()) {
+            cubeSetupFacet.setCubeConfiguration(CubeConfiguration.OPENSHIFT);
+        }
+
+        this.cubeSetupFacet = cubeSetupFacet;
+    }
+
+    private void setCubeConfigurationParamteres(UIExecutionContext context) throws IOException {
+        cubeSetupFacet.setConfigurationParameters(getParametersForExtensionConfiguration(context));
+    }
+
+    private boolean isDockerMachine() {
+        return dockerMachine.hasValue() && dockerMachine.getValue();
+    }
+
+    private boolean isDockerOrDockerCompose() {
+        return isDocker() || isDockerCompose();
+    }
+
+    private boolean isDocker() {
+        return type.hasValue() && "docker".equals(getStringForCLIDisplay(type.getValue()));
```

Comment:
Why do we call `getStringForCLIDisplay` in this case? Shouldn't value be the same from IDE and CLI?